applications/crossbar/src/modules/cb_notifications.erl

```diff
@@ -298,6 +298,23 @@ do_post(Context) ->
         _Status -> Context1
     end.
 
+-spec set_system_macros(cb_context:context()) -> cb_context:context().
+-spec set_system_macros(cb_context:context(), api_binary()) -> cb_context:context().
+set_system_macros(Context) ->
+    Id = wh_doc:id(cb_context:doc(Context)),
+    set_system_macros(Context, Id).
+
+set_system_macros(Context, 'undefined') -> Context;
```

Comment:
This seems unnecessary, no? When would the doc id be missing? Would we really want to continue processing a POST with no id?